pkg/codec/openh264/bridge.cpp

```diff
@@ -51,8 +54,10 @@ Encoder *enc_new(const EncoderOptions opts) {
     errno = rv;
     return NULL;
   }
+  printf("3: %d\n", errno);
```

Comment:
So, somehow the errno got set to 60 (Operation timeout) here. But, when I tried to ignore this errno, it worked just fine.

Reply:
I found that errno is not goroutine safe on Darwin.
```go
package main
  
import (
  "fmt"
  "net"
  "time"
)

// #include <unistd.h>
// #include <errno.h>
// int wait() { usleep(2000000); }
import "C"

func main() {
  for i := 0; i < 10; i++ {
    go func() {
      for {
        _, _ = net.Dial("tcp", "localhost:12345")
        time.Sleep(time.Millisecond)
      }
    }()
  }
  _, err := C.wait()
  fmt.Printf("err: %v\n", err)
}
```
This code outputs `nil` on Linux, but on Darwin (tested on Travis-CI's OSX environment) outputs `operation timed out`.

Reply:
So, it seems unsafe to use errno on Darwin.
How about https://github.com/pion/mediadevices/pull/88?

Reply:
I always thought that the cgo error handling is pretty neat, using the second value for error by implying from errno. But, nevermind, it's not safe.... Oh, well, I like the approach, I don't know if there's a cleaner way of doing it atm.